File: _v3/rowereddots/stock_conform.py
import os, subprocess, re, json, sys
os.chdir("D:/Proyectos/video2-wt/rowereddots")
RAW="_v3/rowereddots/stock_raw"; OUT="public/broll/rowereddots/stock"; os.makedirs(OUT, exist_ok=True)
rep={}
for f in sorted(os.listdir(RAW)):
    if not f.endswith(".mp4") or f.startswith("_"): continue
    src=f"{RAW}/{f}"; dst=f"{OUT}/{f}"
    dur=float(subprocess.check_output(["ffprobe","-v","error","-show_entries","format=duration","-of","csv=p=0",src],text=True).strip().rstrip(","))
    o=subprocess.run(["ffmpeg","-v","info","-t","2","-i",src,"-an","-vf","scale=480:270,signalstats,metadata=print:key=lavfi.signalstats.YAVG","-f","null","-"],capture_output=True,text=True).stderr
    ys=[float(x) for x in re.findall(r"YAVG=([0-9.]+)",o)]
    ss=0.0
    for i,y in enumerate(ys):
        if y>=40: ss=i/ max(1,len(ys)/2.0); break
    else: ss=0.5
    ss=round(min(ss+0.1,1.5),2)
    t=min(12.0, dur-ss-0.1)
    if not os.path.exists(dst):
        subprocess.run(["ffmpeg","-v","error","-y","-ss",str(ss),"-i",src,"-t",str(t),"-an","-vf","scale=1920:1080:force_original_aspect_ratio=increase,crop=1920:1080,fps=30,setsar=1","-c:v","libx264","-crf","20","-preset","veryfast","-pix_fmt","yuv420p",dst],check=True)
    o2=subprocess.run(["ffmpeg","-v","info","-t","0.7","-i",dst,"-an","-vf","signalstats,metadata=print:key=lavfi.signalstats.YAVG","-f","null","-"],capture_output=True,text=True).stderr
    y2=[float(x) for x in re.findall(r"YAVG=([0-9.]+)",o2)]
    rep[f]={"ss":ss,"t":round(t,2),"ymin":round(min(y2),1) if y2 else None}
    print(f, rep[f], flush=True)
print("midió", len(rep), "clips · arranque oscuro (<40):", [k for k,v in rep.items() if (v["ymin"] or 0)<40])

File: work/famarioneta/stock_conform.mjs
import fs from "node:fs";
import { execFileSync } from "node:child_process";
const P = JSON.parse(fs.readFileSync("_v3/famarioneta_stock_pick.json", "utf8"));
const probe = (f) => parseFloat(execFileSync("ffprobe", ["-v", "error", "-show_entries", "format=duration", "-of", "csv=p=0", f], { encoding: "utf8" }));
const luma = (f, ss) => { const o = execFileSync("ffmpeg", ["-v", "info", "-ss", String(ss), "-t", "0.7", "-i", f, "-an", "-vf", "scale=320:180,signalstats,metadata=print:key=lavfi.signalstats.YAVG", "-f", "null", "-"], { encoding: "utf8", stdio: ["ignore", "pipe", "pipe"] }); return 0; };
let n = 0;
for (const [name, p] of Object.entries(P)) {
  const out = `public/broll/famarioneta/stock/${name}.mp4`;
  if (fs.existsSync(out)) { n++; continue; }
  const d = probe(p.raw);
  const ss = d > 9 ? 1.0 : 0.4;
  const t = Math.min(14, d - ss - 0.3);
  execFileSync("ffmpeg", ["-v", "error", "-y", "-ss", String(ss), "-i", p.raw, "-t", t.toFixed(2), "-an", "-vf", "scale=1920:1080:force_original_aspect_ratio=increase,crop=1920:1080,fps=30,setsar=1,format=yuv420p", "-c:v", "libx264", "-preset", "veryfast", "-crf", "21", "-g", "30", "-r", "30", out]);
  n++; console.log(name, t.toFixed(1));
}
import { spawnSync } from "node:child_process";
const malos = []; let medidos = 0;
for (const name of Object.keys(P)) {
  const f = `public/broll/famarioneta/stock/${name}.mp4`;
  const r = spawnSync("ffmpeg", ["-v", "info", "-t", "0.7", "-i", f, "-an", "-vf", "signalstats,metadata=print:key=lavfi.signalstats.YAVG", "-f", "null", "-"], { encoding: "utf8" });
  const v = [...(r.stderr || "").matchAll(/YAVG=([\d.]+)/g)].map((m) => +m[1]);
  if (!v.length) continue; medidos++;
  if (Math.min(...v) < 40) malos.push(name + " " + Math.min(...v).toFixed(0));
}
console.log(`luma arranque: medidos ${medidos}/${Object.keys(P).length} · <40: ${malos.join(", ") || "ninguno"}`);
spawnSync("ffmpeg", ["-v", "error", "-y", ...Object.keys(P).flatMap((n) => ["-ss", "3", "-i", `public/broll/famarioneta/stock/${n}.mp4`]), "-filter_complex", Object.keys(P).map((n, i) => `[${i}:v]scale=320:180,trim=end_frame=1[v${i}]`).join(";") + ";" + Object.keys(P).map((n, i) => `[v${i}]`).join("") + `xstack=inputs=${Object.keys(P).length}:layout=${Object.keys(P).map((n, i) => `${(i % 8) * 320}_${Math.floor(i / 8) * 180}`).join("|")}`, "-frames:v", "1", "work/famarioneta/stock_sheet.jpg"]);
console.log("MEDIDO conformados", n, "de", Object.keys(P).length, Object.keys(P).join(" "));
